plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/components/upgrade-product-image-gallery/index.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+import { createBlock, BlockInstance } from '@wordpress/blocks';
+import { dispatch, useSelect } from '@wordpress/data';
+import { UpgradeDowngradeNotice } from '@woocommerce/editor-components/upgrade-downgrade-notice';
+import { findBlock } from '@woocommerce/utils';
+
+const upgradeToBlockifiedProductGallery = (
+	productImageGalleryBlockClientId: string
+) => {
+	const newBlock = createBlock( 'woocommerce/product-gallery' );
+
+	dispatch( 'core/block-editor' ).replaceBlock(
+		productImageGalleryBlockClientId,
+		newBlock
+	);
+
+	return true;
+};
+
+export const UpgradeProductImageGallery = () => {
+	const [ productImageGalleryBlock, setProductImageGalleryBlock ] =
+		useState< BlockInstance | null >( null );
+
+	const getBlocks = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).getBlocks;
+	}, [] );
+
+	useEffect( () => {
+		const foundBlock = findBlock( {
+			blocks: getBlocks(),
+			findCondition: ( block ) =>
+				block.name === 'woocommerce/product-image-gallery',
+		} );
+		if ( foundBlock ) {
+			setProductImageGalleryBlock( foundBlock );
+		} else {
+			setProductImageGalleryBlock( null );
+		}
+	}, [ getBlocks, setProductImageGalleryBlock ] );
+
+	if ( ! productImageGalleryBlock ) {
+		return null;
+	}
+
+	const notice = __(
+		'This template contains the classic Product Image Gallery block which is not compatible with the Add to Cart + Options block. Switch to the new Product Gallery block for a better experience.',
```

Comment:
```suggestion
		'This template contains the classic Product Image Gallery block which is not compatible with the Add to Cart with Options block. Switch to the new Product Gallery block for a better experience.',
```

Reply:
Also in DSGWOO-268, it was decided the block name would be _Add to Cart + Options_ block.

In code, slugs, etc. we use `add-to-cart-with-options`/`addToCartWithOptions` to avoid the `+` sign messing things up, but in user-facing strings we use _Add to Cart + Options_.